artemis-commons/src/main/java/org/apache/activemq/artemis/utils/critical/CriticalAnalyzerImpl.java

```diff
@@ -31,7 +34,27 @@
 
    private volatile long timeoutNanoSeconds;
 
-   private volatile long checkTimeNanoSeconds;
+   // one minute by default.. the server will change it for sure
+   private volatile long checkTimeNanoSeconds = TimeUnit.SECONDS.toNanos(60);
+
+   private final ActiveMQScheduledComponent scheduledComponent;
+
+   private final AtomicBoolean running = new AtomicBoolean(false);
+
+   public CriticalAnalyzerImpl() {
+      // this will make the scheduled component to start its own pool
+      this(null, null);
+   }
+
+   public CriticalAnalyzerImpl(ScheduledExecutorService scheduledExecutorService, Executor executor) {
```

Comment:
Should this then become private? To avoid people using it withou the null's?

Reply:
I can remove the constructor.. and document the intention.

Reply:
removed